frontend/web/src/features/bookingengine/listingdetails/views/amenitiesContainer.js

```diff
@@ -1,22 +1,76 @@
+import { useState } from "react";
 import Amenities from "../../../../store/amenities";
 import Amenity from "../components/amenity";
+import { ALLOWED_AMENITY_CATEGORIES } from "../store/amenityCategories";
 
-const AmenitiesContainer = ({amenityIds}) => {
-    const amenities = amenityIds.map(amenity =>
-        Amenities.find(amenitiesEntry => amenitiesEntry.id === amenity.amenityId)
-    );
-    const featuredAmenities = amenities.filter(amenity => amenity.category === "Essentials");
-
-    return (
-      <div className="amenities-container">
-        <p className="amenities-title">This place offers the following:</p>
-        <div className="essential-amenities-container">
-          {featuredAmenities.map((amenity) => {
-            return <Amenity key={amenity.id} amenity={amenity} />;
-          })}
-        </div>
+const AmenitiesContainer = ({ amenityIds }) => {
+  const [showModal, setShowModal] = useState(false);
+
+  const amenityObjects = amenityIds.map((a) =>
+    Amenities.find((entry) => entry.id === a.amenityId)
+  );
+
+  // FEATURED (first small section)
+  const featured = amenityObjects.filter((a) =>
+    ALLOWED_AMENITY_CATEGORIES.includes(a?.category)
+  );
+
+  // GROUP ALL FOR MODAL
+  const grouped = {};
+  amenityObjects.forEach((item) => {
+    if (!item) return;
+    if (!grouped[item.category]) grouped[item.category] = [];
+    grouped[item.category].push(item);
+  });
+
+  return (
+    <div className="amenities-container">
+      <p className="amenities-title">This place offers the following:</p>
+
+      {/* ⭐ FEATURED GRID */}
+      <div className="amenities-featured">
+        {featured.slice(0, 6).map((amenity) => (
+          <Amenity key={amenity.id} amenity={amenity} />
+        ))}
       </div>
-    );
+
+      {/* ⭐ BUTTON */}
+      <button 
+        className="show-all-amenities-btn"
+        onClick={() => setShowModal(true)}
+      >
+        Toon alle voorzieningen
+      </button>
+
+      {/* ⭐ MODAL */}
+      {showModal && (
+        <div className="amenities-modal-overlay" onClick={() => setShowModal(false)}>
+          <div className="amenities-modal" onClick={(e) => e.stopPropagation()}>
+            
+            <button className="close-modal-btn" onClick={() => setShowModal(false)}>×</button>
+
+            <h2 className="modal-title">Voorzieningen</h2>
+
+            {/* GROUPED LIST */}
```

Comment:
'List' is quite ambiguous, so consider this:
 ```suggestion
              {/* GROUPED AMENITIES*/}
```